### PR TITLE
Refactor 'Battery Status'

### LINKED ISSUE #49: Battery Status - Attempted to Divide by 0
This scanner keeps failing with "attempting to divide by 0". It would be nice to either throw a string that says the battery is bad or at least have 0 as the value

### PR DESCRIPTION
**Pull request by ColbyBouma**
Fixes #49

While fixing #49, I decided to refactor this script a bit. Replace Get-WmiObject with Get-CimInstance, remove the unnecessary custom function, etc.

I don't have any Windows laptops to test this on, so please test it before approving :)

### CODE REVIEW

File: PowerShell Scanners/Battery Status/Battery Status.ps1 [hunk 1]
```diff
@@ -1,23 +1,33 @@
-﻿function Format-Battery($BatteryName = "No Battery Detected", $DesignedCapacity = 0, $FullCharge = 0, $Health = 0) {
+﻿$Batteries = (Get-CimInstance -ClassName "BatteryStatus" -Namespace "ROOT\WMI" -ErrorAction SilentlyContinue)
+$AllBatteryData = Get-CimInstance -ClassName "BatteryStaticData" -Namespace "ROOT\WMI"
```
Comment (sstodd7532) [inline PowerShell Scanners/Battery Status/Battery Status.ps1]: I get a `Get-CimInstance: Generic failure` on this line

File: PowerShell Scanners/Battery Status/Battery Status.ps1 [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+$AllCharges = (Get-CimInstance -ClassName "BatteryFullChargedCapacity" -Namespace "ROOT\WMI").FullChargedCapacity
+$BatteryIndex = 0
+
+Foreach ($Battery in $Batteries) {
+
+    $BatteryName = $AllBatteryData.DeviceName.split(" ")[$BatteryIndex]
```
Comment (sstodd7532) [inline PowerShell Scanners/Battery Status/Battery Status.ps1:8]: This line fails cause of the issues with line 2

File: PowerShell Scanners/Battery Status/Battery Status.ps1 [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+    $DesignedCapacity = $AllBatteryData.DesignedCapacity[$BatteryIndex]
```
Comment (sstodd7532) [inline PowerShell Scanners/Battery Status/Battery Status.ps1:9]: Same issue here. Line 2 failed and has no data to get from the array.

File: PowerShell Scanners/Battery Status/Battery Status.ps1 (hunk 1 above)
Comment (ColbyBouma) [reply PowerShell Scanners/Battery Status/Battery Status.ps1]: Does the old script work for you?

Comment (sstodd7532) [reply PowerShell Scanners/Battery Status/Battery Status.ps1]: Yes it did with PowerShell 5.1. 7.1.2 and 5.1 get the same results with the changes that you made.

Comment (jbullock1) [reply PowerShell Scanners/Battery Status/Battery Status.ps1]: function Format-Battery($BatteryName = "No Battery Detected", $DesignedCapacity = 0, $FullCharge = 0, $Health = 0) {
    [PSCustomObject]@{
        Name             = $BatteryName
        DesignedCapacity = $DesignedCapacity
        FullCharge       = $FullCharge
        Health           = $Health
    }
}


$Batteries = (Get-WmiObject -Class "BatteryStatus" -Namespace "ROOT\WMI" -ErrorAction SilentlyContinue)
$AllBatteryData = Get-WmiObject -Class "BatteryStaticData" -Namespace "ROOT\WMI"
$AllCharges = (Get-WmiObject -Class "BatteryFullChargedCapacity" -Namespace "ROOT\WMI").FullChargedCapacity
$BatteryIndex = 0
foreach ($Battery in $Batteries) {
    $BatteryName = $AllBatteryData.DeviceName.split(" ")[$BatteryIndex]
    $DesignedCapacity = $AllBatteryData.DesignedCapacity[$BatteryIndex]
    $FullCharge = $AllCharges[$BatteryIndex]
    $Health = [math]::Round($FullCharge / $DesignedCapacity * 100, 2)

    Format-Battery $BatteryName $DesignedCapacity $FullCharge $Health
    $BatteryIndex++
}

returns this for me from original post

Name DesignedCapacity FullCharge Health
---- ---------------- ---------- ------
DELL            60002      55054  91.75

Comment (ColbyBouma) [reply PowerShell Scanners/Battery Status/Battery Status.ps1]: I don't know why, but `Get-CimInstance` just doesn't work with `BatteryStaticData`. I changed it back to `Get-WmiObject`. Also, I installed Windows on a laptop and this script is working on it now. I should have done that the first time :)

**Review by sstodd7532 — APPROVED** [review]

**Pull request merged**